### PR TITLE
Sensor generic using Simple Streaming Interface (SSI)

### PR DESCRIPTION
**Pull request by murthyvedula**
This pull request adds generic sensor code for easy adaptation by end users to add or replace with a different sensor.
The example code uses onboard mCube accelerometer sensor to illustrate the concept and usage.

### CODE REVIEW

File: qf_apps/qf_ssi_ai_app/GCC_Project/config.mk [hunk 1]
```diff
@@ -162,9 +163,10 @@ $(info ######  ERROR - TC_PATH is not defined in config.mk #########)
 exit
 endif
 export NM="$(TC_PATH)/arm-none-eabi-nm"
-export LD="$(TC_PATH)/arm-none-eabi-gcc"
+export LD="$(TC_PATH)/arm-none-eabi-g++"
 export AS="$(TC_PATH)/arm-none-eabi-gcc" -c
 export CC="$(TC_PATH)/arm-none-eabi-gcc" -c
+export CPLUSPLUS="$(TC_PATH)\arm-none-eabi-g++" -c
```
Comment (jublin) [inline qf_apps/qf_ssi_ai_app/GCC_Project/config.mk:169]: this needs to be a forward slash for linux builds.

**Review by timsaxe — APPROVED** [review]
Looks fine to me

File: qf_apps/qf_ssi_ai_app/GCC_Project/config.mk (hunk 1 above)
Comment (murthyvedula) [reply qf_apps/qf_ssi_ai_app/GCC_Project/config.mk]: updated to fix the path issue

**Comment by jublin** [comment]
This does not correctly run with the specified ARM-gcc version in the README. 

qf_ssi_ai_app hangs after "Hello World!!" when compiled with arm-gcc 9.3.1 as well as 9.2.1.

**Comment by jublin** [comment]
Latest change to branch fixes issue with arm-gcc 9.x

**Pull request merged**